src/AssetForm/containers/DraftInput.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Containers which connect fields in the draft to input fields.
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { patchDraft } from '../../redux/actions/editAsset';
+
+/**
+ * A container component which wraps an input control connecting it to a field in the draft. By
+ * default it wraps a HTML input element but this can be overridden with the "component" prop.
+ *
+ * The mapDraftToInputProps and mapOnChangeToPatch determine how the draft is mapped to and from
+ * the input props. They are both functions.
+ *
+ * mapDraftToInputProps takes the component props and current draft and returns an object which
+ * will be used to set the props on the wrapped input control.
+ *
+ * mapOnChangeToPatch is passed the component props as its first argument and then the arguments
+ * passed to the onChange handler for the wrapper input. It should return a value suitable for
+ * passing to the patchDraft action.
+ *
+ * By convention, the "name" prop specifies which field in the draft should be connected although
+ * technically mapDraftToInputProps and mapOnChangeToPatch could use whichever props they wish.
+ */
+class UnconnectedDraftInput extends Component {
+  handleChange = (...args) => {
+    const { patchDraft, mapOnChangeToPatch } = this.props;
+    patchDraft(mapOnChangeToPatch(this.props, ...args));
+  }
+
+  render = () => {
+    // We swallow map{...} and patchDraft to avoid passing them through to the wrapped component.
+    const {
+      component: Component,
+      mapDraftToInputProps,
+      mapOnChangeToPatch,
+      patchDraft,
+      children,
+      ...rest
+    } = this.props;
+
+    return <Component {...rest} onChange={this.handleChange}>{ children }</Component>;
+  }
+};
+
+UnconnectedDraftInput.propTypes = {
+  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  mapOnChangeToPatch: PropTypes.func.isRequired,
+};
+
+UnconnectedDraftInput.defaultProps = {
+  component: 'input',
+};
+
+const mapStateToProps = ({ editAsset: { draft } }, ownProps) => {
+  return ownProps.mapDraftToInputProps(draft, ownProps);
+};
+
+export const DraftInput = connect(mapStateToProps, { patchDraft })(UnconnectedDraftInput);
+
+DraftInput.propTypes = {
+  mapDraftToInputProps: PropTypes.func.isRequired,
+};
+
+export default DraftInput;
+
+/**
+ * Version of DraftInput pre-configured to work with <input type="checkbox"> components. Accesses a
```

Comment:
This is a copy-and-paste bug. It should read "input components" and not mention checkboxen.